celements-scheduler/src/main/java/com/celements/scheduler/classdefs/SchedulerJobClass.java

```diff
@@ -0,0 +1,92 @@
+package com.celements.scheduler.classdefs;
+
+import org.springframework.stereotype.Component;
+import org.xwiki.model.reference.ClassReference;
+
+import com.celements.marshalling.EnumMarshaller;
+import com.celements.model.classes.AbstractClassDefinition;
+import com.celements.model.classes.fields.ClassField;
+import com.celements.model.classes.fields.CustomStringField;
+import com.celements.model.classes.fields.LargeStringField;
+import com.celements.model.classes.fields.StringField;
+import com.xpn.xwiki.XWikiConstant;
+
+@Component(SchedulerJobClass.CLASS_DEF_HINT)
+public class SchedulerJobClass extends AbstractClassDefinition
+    implements SchedulerJobClassDefinition {
+
+  public static final String DOC_NAME = "SchedulerJobClass";
+  public static final String XWIKI_SPACE = XWikiConstant.XWIKI_SPACE;
+  public static final String CLASS_DEF_HINT = XWIKI_SPACE + "." + DOC_NAME;
+  public static final ClassReference CLASS_REF = new ClassReference(XWIKI_SPACE, DOC_NAME);
+
+  public enum Status {
+    NORMAL, NONE
+  }
+
+  public static final ClassField<String> FIELD_JOB_NAME = new StringField.Builder(
+      CLASS_REF, "jobName")
+          .prettyName("Job Name")
+          .size(60)
+          .build();
+
+  public static final ClassField<String> FIELD_JOB_DESCRIPTION = new LargeStringField.Builder(
+      CLASS_REF, "jobDescription")
+          .rows(10)
+          .prettyName("Job Description")
+          .size(45)
+          .build();
+
+  public static final ClassField<String> FIELD_JOB_CLASS = new StringField.Builder(
+      CLASS_REF, "jobClass")
+          .prettyName("Job Class")
+          .size(60)
+          .build();
+
+  public static final ClassField<Status> FIELD_STATUS = new CustomStringField.Builder<>(
```

Comment:
Why not using `CustomSingleListField`? This would possibly change the property to a static-list property. Yet, this is still accessible like a string field and hence cannot break any code.

Reply:
@fpichler I'm afraid this would break existing data. when the class is recreated, it creates a static-list property and will thus lose the connection to already existing string values stored in a different table than static-list values (the class isn't mapped).

changing an unmapped string property to a single list field requires a migration, so we should keep it as a CustomStringField for now.